pkg/s3select/input.go

```diff
@@ -0,0 +1,379 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package s3select
+
+import (
+	"bytes"
+	"compress/bzip2"
+	"encoding/csv"
+	"encoding/xml"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"net/http"
+
+	gzip "github.com/klauspost/pgzip"
+)
+
+const progressTime time.Duration = 60 * time.Second
+const continuationTime time.Duration = 5 * time.Second
+
+var curBuf *bytes.Buffer
+var progressTicker *time.Ticker
```

Comment:
why are you keeping these as global variables? is there any problem without having them as local variables?